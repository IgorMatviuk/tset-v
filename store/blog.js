export const mutationTypes = {
  SET_LOADING_START: 'SET_LOADING_START',
  SET_LOADING_SUCCESS: 'SET_LOADING_SUCCESS',
  SET_CURRENT_PAGE: 'SET_CURRENT_PAGE'
}

export const actionsTypes = {
  GET_BLOG_POSTS: 'GET_BLOG_POSTS'
}

export const state = () => ({
  posts: [],
  isLoading: false,
  currentPage: null,
  totalPages: null
})

export const mutations = {
  [mutationTypes.SET_LOADING_START](state, status) {
    state.isLoading = status
  },
  [mutationTypes.SET_CURRENT_PAGE](state, currentPage) {
    state.currentPage = currentPage
  },
  [mutationTypes.SET_LOADING_SUCCESS](
    state,
    { posts, currentPage, totalPages }
  ) {
    state.posts = posts
    state.currentPage = currentPage
    state.totalPages = totalPages

    if (state.isLoading) {
      state.isLoading = false
    }
  }
}

export const actions = {
  async [actionsTypes.GET_BLOG_POSTS]({ commit }, params) {
    commit(mutationTypes.SET_LOADING_START, true)

    let query = Object.keys(params)
      .map((key) => `${key}=${encodeURI(params[key])}`)
      .join('&')

    if (!params.limit) {
      query += '&limit=8'
    }

    const { data, meta } = await this.$axios.$get(`api/posts?${query}`)

    const totalPages = Math.ceil(meta.total / (params.limit || 8))

    commit(mutationTypes.SET_LOADING_SUCCESS, {
      posts: data,
      currentPage: meta.current_page,
      totalPages
    })
  }
}
