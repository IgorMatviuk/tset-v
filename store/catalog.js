import { CATALOG_ITEMS_PER_PAGE } from '@/utils/constants'

function createRootCategories(categories) {
  const withRootList = categories.map((category) => {
    if (category.name_alias) {
      const rootCategory = {
        id: category.id,
        name: category.name_alias,
        slug: category.slug
      }

      category.categories = [
        rootCategory,
        ...createRootCategories(category.categories)
      ]
    }

    return category
  })

  return withRootList
}

export const mutationTypes = {
  SET_CATEGORIES: 'SET_CATEGORIES',
  SET_ACTIVE_CATEGORY: 'SET_ACTIVE_CATEGORY',
  SET_PRODUCTS: 'SET_PRODUCTS',
  SET_ATTRIBUTES: 'SET_ATTRIBUTES',
  SET_FILTERS: 'SET_FILTERS',
  CLEAR_FILTERS: 'CLEAR_FILTERS',
  SET_CATALOG_DATA: 'SET_CATALOG_DATA',
  SET_ORDER: 'SET_ORDER',
  SET_SHOW_ITEMS_COUNT: 'SET_SHOW_ITEMS_COUNT',
  SET_LOADING_TOGGLE: 'SET_LOADING_TOGGLE',
  SET_CURRENT_PAGE: 'SET_CURRENT_PAGE',
  SET_TOTAL_PAGES: 'SET_TOTAL_PAGES',
  RESET_META_DATA: 'RESET_META_DATA',
  SET_SEO_TEXT: 'SET_SEO_TEXT'
}

export const actionTypes = {
  GET_CATEGORIES: 'GET_CATEGORIES',
  GET_PRODUCTS: 'GET_PRODUCTS',
  GET_ATTRIBUTES: 'GET_ATTRIBUTES'
}

export const state = () => ({
  categories: [],
  activeCategory: null,
  products: [],
  attributes: [],
  filters: [],
  order: 'popularity',
  showItemsCount: 4,
  isLoading: false,
  currentPage: null,
  totalPages: null,
  seoText: ''
})

export const mutations = {
  [mutationTypes.SET_CATEGORIES](state, categories) {
    state.categories = categories
  },
  [mutationTypes.SET_ACTIVE_CATEGORY](state, activeCategory) {
    state.activeCategory = activeCategory
  },
  [mutationTypes.SET_PRODUCTS](state, products) {
    state.products = products
  },
  [mutationTypes.SET_ATTRIBUTES](state, attributes) {
    state.attributes = attributes
  },
  [mutationTypes.SET_CATALOG_DATA](state, { category, products, attributes }) {
    state.activeCategory = category
    state.products = products
    state.attributes = attributes
  },
  [mutationTypes.SET_ORDER](state, order) {
    state.order = order
  },
  [mutationTypes.SET_SHOW_ITEMS_COUNT](state, showItemsCount) {
    state.showItemsCount = showItemsCount
  },
  [mutationTypes.SET_FILTERS](state, filters) {
    state.filters = filters
  },
  [mutationTypes.CLEAR_FILTERS](state) {
    state.order = 'popularity'
    state.filters = []
  },
  [mutationTypes.SET_LOADING_TOGGLE](state, status) {
    state.isLoading = status
  },
  [mutationTypes.SET_CURRENT_PAGE](state, currentPage) {
    state.currentPage = currentPage
  },
  [mutationTypes.SET_TOTAL_PAGES](state, totalPages) {
    state.totalPages = totalPages
  },
  [mutationTypes.RESET_META_DATA](state) {
    state.filters = []
    state.order = 'popularity'
    state.currentPage = 1
  },
  [mutationTypes.SET_SEO_TEXT](state, seoText) {
    state.seoText = seoText
  }
}

export const actions = {
  async [actionTypes.GET_CATEGORIES]({ commit }) {
    // FIXME: temporary solution
    const STAGE_API_URL = 'https://back.virna.com.ua'
    const { data: categories } = await this.$axios.$get(
      `${STAGE_API_URL}/api/categories`
    )

    // const { data: categories } = await this.$axios.$get('/api/categories')
    const normilizedCategories = createRootCategories(categories)

    commit(mutationTypes.SET_CATEGORIES, normilizedCategories)
  },
  async [actionTypes.GET_PRODUCTS]({ commit }, { url, query }) {
    commit(mutationTypes.SET_LOADING_TOGGLE, true)

    // FIXME: temporary solution
    const STAGE_API_URL = 'https://back.virna.com.ua'
    const {
      data: { products },
      meta
    } = await this.$axios.$get(
      `${STAGE_API_URL}/api/router?url=${url}${query ? `&${query}` : ''}`
    )

    const totalPages = Math.ceil(meta.total / CATALOG_ITEMS_PER_PAGE)

    commit(mutationTypes.SET_PRODUCTS, products)
    commit(mutationTypes.SET_CURRENT_PAGE, meta.current_page)
    commit(mutationTypes.SET_TOTAL_PAGES, totalPages)
    commit(mutationTypes.SET_LOADING_TOGGLE, false)
  }
}

export const getters = {
  defaultCategorySlug(state) {
    // FIXME: temporary solution
    return state.categories[0].slug
  },
  optionLabel(state) {
    const { attributes } = state

    return (code, slug) => {
      const { options } = attributes.find((attr) => attr.code === code)
      const { label } = options.find((option) => option.slug === slug)
      return label
    }
  },
  selectedAttributeOptions(state) {
    const { filters } = state

    return (code) => {
      const filter = filters.find((filter) => filter.code === code)
      return filter ? filter.value : null
    }
  }
}
