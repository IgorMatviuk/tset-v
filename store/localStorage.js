export const mutationTypes = {
  SET_CART_ID: 'SET_CART_ID'
}

export const state = () => ({
  cartId: null,
  status: false
})

export const mutations = {
  [mutationTypes.SET_CART_ID](state, cartId) {
    state.cartId = cartId
  }
}
