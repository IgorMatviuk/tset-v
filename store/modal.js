export const mutationTypes = {
  SET_MODAL_META: 'SET_MODAL_META',
  SET_MODAL_VISIBILITY_TOGGLE: 'SET_MODAL_VISIBILITY_TOGGLE',
  SET_MODAL_DATA: 'SET_MODAL_DATA',
  RESET_MODAL_STATE: 'RESET_MODAL_STATE'
}

function createInitialState() {
  return {
    form: {
      visible: false,
      meta: {
        name: '',
        title: '',
        fields: [],
        actionText: '',
        submitText: '',
        links: [],
        action: {}
      },
      data: {}
    },
    message: {
      visible: false,
      meta: {
        title: '',
        content: ''
      }
    },
    testimonials: {
      visible: false,
      data: {}
    }
  }
}

export const state = () => createInitialState()

export const mutations = {
  [mutationTypes.SET_MODAL_META](state, payload) {
    const { type, ...meta } = payload

    if (type === 'form' && meta.success) {
      state.message.meta = meta.success
    }

    state[type].meta = meta
  },
  [mutationTypes.SET_MODAL_VISIBILITY_TOGGLE](state, { type, status }) {
    state[type].visible = status
  },
  [mutationTypes.SET_MODAL_DATA](state, { type, data }) {
    state[type].data = data
  },
  [mutationTypes.RESET_MODAL_STATE](state) {
    const { form, message, testimonials } = createInitialState()

    state.form = form
    state.message = message
    state.testimonials = testimonials
  }
}
