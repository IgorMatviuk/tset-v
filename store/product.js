export const mutationTypes = {
  SET_PRODUCT_DATA: 'SET_PRODUCT_DATA',
  SET_ACTIVE_VARIANT: 'SET_ACTIVE_VARIANT',
  SET_SECTIONS_DATA: 'SET_SECTIONS_DATA',
  SET_NAV_LINKS: 'SET_NAV_LINKS',
  SET_SELECTED_SIZE: 'SET_SELECTED_SIZE',
  SET_SELECTED_COLOR: 'SET_SELECTED_COLOR',
  SET_IMAGE_VIEWER_VISIBILITY_TOGGLE: 'SET_IMAGE_VIEWER_VISIBILITY_TOGGLE',
  SET_ACTIVE_PREVIEW: 'SET_ACTIVE_PREVIEW'
}

export const state = () => ({
  productData: {},
  sectionsData: {},
  activeVariant: {},
  links: {},
  selectedSize: null,
  selectedColor: null,
  imageViewerVisibility: false,
  activePreview: {}
})

export const mutations = {
  [mutationTypes.SET_PRODUCT_DATA](state, productData) {
    state.productData = productData
  },
  [mutationTypes.SET_ACTIVE_VARIANT](state, activeVariant) {
    state.activeVariant = activeVariant
  },
  [mutationTypes.SET_SECTIONS_DATA](state, sectionsData) {
    state.sectionsData = sectionsData
  },
  [mutationTypes.SET_NAV_LINKS](state, links) {
    state.links = links
  },
  [mutationTypes.SET_SELECTED_SIZE](state, size) {
    state.selectedSize = size
  },
  [mutationTypes.SET_SELECTED_COLOR](state, color) {
    state.selectedColor = color
  },
  [mutationTypes.SET_IMAGE_VIEWER_VISIBILITY_TOGGLE](state, status) {
    state.imageViewerVisibility = status
  },
  [mutationTypes.SET_ACTIVE_PREVIEW](state, preview) {
    state.activePreview = preview
  }
}

export const getters = {
  seoText(state) {
    return state.productData.attribute_family.description.description.value
  }
}
