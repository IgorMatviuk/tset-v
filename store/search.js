export const mutationTypes = {
  SET_LOADING_START: 'SET_LOADING_START',
  SET_LOADING_SUCCESS: 'SET_LOADING_SUCCESS',
  SET_SEARCH_QUERY: 'SET_SEARCH_QUERY',
  SET_CURRENT_PAGE: 'SET_CURRENT_PAGE',
  SET_TOTAL_PAGES: 'SET_TOTAL_PAGES'
}

export const actionTypes = {
  GET_PRODUCTS: 'GET_PRODUCTS'
}

export const state = () => ({
  products: [],
  query: '',
  isLoading: false,
  currentPage: null,
  totalPages: null
})

export const mutations = {
  [mutationTypes.SET_LOADING_START](state) {
    state.isLoading = true
  },
  [mutationTypes.SET_LOADING_SUCCESS](state, products) {
    state.products = products
    state.isLoading = false
  },
  [mutationTypes.SET_SEARCH_QUERY](state, query) {
    state.query = query
  },
  [mutationTypes.SET_CURRENT_PAGE](state, currentPage) {
    state.currentPage = currentPage
  },
  [mutationTypes.SET_TOTAL_PAGES](state, totalPages) {
    state.totalPages = totalPages
  }
}

export const actions = {
  async [actionTypes.GET_PRODUCTS]({ commit }, { query, page }) {
    commit(mutationTypes.SET_LOADING_START, true)

    const encodedQuery = !/^(%[0-9a-f]{2})+$/.test(query)
      ? encodeURIComponent(query)
      : query

    let queryStr = `q=${encodedQuery}`

    if (page) {
      queryStr += `&page=${page}`
    }

    // FIXME: temporary solution
    const STAGE_API_URL = 'https://back.virna.com.ua'
    const { data: products, meta } = await this.$axios.$get(
      `${STAGE_API_URL}/api/search?${queryStr}`
    )

    commit(mutationTypes.SET_LOADING_SUCCESS, products)
    commit(mutationTypes.SET_SEARCH_QUERY, query)
    commit(mutationTypes.SET_CURRENT_PAGE, meta.current_page)
    commit(mutationTypes.SET_TOTAL_PAGES, meta.total)
  }
}
