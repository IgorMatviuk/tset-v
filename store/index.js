import { actionTypes as catalogActionTypes } from '@/store/catalog'

export const mutationTypes = {
  SET_LANG: 'SET_LANG',
  SET_MENUS: 'SET_MENUS',
  SET_PAGE_TYPE: 'SET_PAGE_TYPE',
  SET_BREADCRUMB: 'SET_BREADCRUMB',
  SET_CONTACTS: 'SET_CONTACTS',
  SET_GLOBALS: 'SET_GLOBALS'
}

export const actionTypes = {
  GET_MENUS: 'GET_MENUS',
  GET_CONTACTS: 'GET_CONTACTS',
  GET_BANNER: 'GET_BANNER',
  GET_CLIENT_LOCATION: 'GET_CLIENT_LOCATION'
}

export const state = () => {
  return {
    locale: 'ru',
    locales: ['uk', 'ru'],
    fallbackLocale: 'ru',
    pageType: null,
    breadcrumb: [],
    menus: [],
    globals: {
      countryCode: 'UA',
      banner: null
    },
    contacts: {
      phone: '0 800 332 512',
      email: 'office@virna.com.ua',
      schedule: 'с 9:00 - 18:00',
      socialLinks: [
        { name: 'instagram', url: 'https://www.instagram.com/' },
        { name: 'facebook', url: 'https://facebook.com/' },
        { name: 'youtube', url: 'https://www.youtube.com/' }
      ]
    }
  }
}

export const mutations = {
  [mutationTypes.SET_LANG](state, locale) {
    state.locale = locale
  },
  [mutationTypes.SET_MENUS](state, menus) {
    state.menus = menus
  },
  [mutationTypes.SET_PAGE_TYPE](state, pageType) {
    state.pageType = pageType
  },
  [mutationTypes.SET_BREADCRUMB](state, breadcrumb) {
    state.breadcrumb = breadcrumb
  },
  [mutationTypes.SET_CONTACTS](state, contacts) {
    state.contacts = contacts
  },
  [mutationTypes.SET_GLOBALS](state, { key, value }) {
    state.globals[key] = value
  }
}

export const actions = {
  async [actionTypes.GET_MENUS]({ commit }) {
    const STAGE_API_URL = 'https://back.virna.com.ua'
    const { data: menus } = await this.$axios.$get(`${STAGE_API_URL}/api/menus`)
    commit(mutationTypes.SET_MENUS, menus)
  },
  async [actionTypes.GET_CONTACTS]({ commit }) {
    const STAGE_API_URL = 'https://back.virna.com.ua'
    const { data: contacts } = await this.$axios.$get(
      `${STAGE_API_URL}/api/contacts`
    )
    commit(mutationTypes.SET_CONTACTS, contacts)
  },
  async [actionTypes.GET_BANNER]({ commit }) {
    const { data: banner } = await this.$axios.$get('/api/banner')
    const defaultBanner = {
      key: 'banner',
      value: banner
    }

    commit(mutationTypes.SET_GLOBALS, defaultBanner)
  },
  async [actionTypes.GET_CLIENT_LOCATION]({ commit }) {
    const locationString = await this.$axios.$get('https://ip2c.org/s')
    const [, countryCode] = locationString.split(';')
    const defaultCountry = {
      key: 'countryCode',
      value: countryCode
    }

    commit(mutationTypes.SET_GLOBALS, defaultCountry)
  },
  async nuxtServerInit({ dispatch }) {
    await Promise.all([
      dispatch(actionTypes.GET_MENUS),
      dispatch(actionTypes.GET_CONTACTS),
      dispatch(`catalog/${catalogActionTypes.GET_CATEGORIES}`)
    ])
  }
}

/**
 * Get menum by type
 * @param {object} state
 * @param {('header'|'footer'|'catalog'|'promo')} type
 */
function getMenuByType(state, type) {
  const { menus } = state
  return menus.find((menu) => menu.menu_type === type)
}

export const getters = {
  mobileMenu(state) {
    return getMenuByType(state, 'mobile')
  },
  desktopMenu(state) {
    return getMenuByType(state, 'header')
  },
  promoMenu(state) {
    return getMenuByType(state, 'promo')
  },
  topMenu(state) {
    return getMenuByType(state, 'top')
  },
  catalogMenu(state) {
    return getMenuByType(state, 'catalog')
  },
  footerMenu(state) {
    return getMenuByType(state, 'footer')
  }
}
