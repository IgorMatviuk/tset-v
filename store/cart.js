import { mutationTypes as localStorageMutationTypes } from '@/store/localStorage'

export const mutationTypes = {
  SET_LOADING_START: 'SET_LOADING_START',
  SET_LOADING_SUCCESS: 'SET_LOADING_SUCCESS',
  SET_CART_DATA: 'SET_CART_DATA',
  SET_COUNTRIES: 'SET_COUNTRIES',
  SET_REGIONS: 'SET_REGIONS',
  SET_STATES: 'SET_STATES',
  SET_ORDER_SAVE: 'SET_ORDER_SAVE',
  CLEAR_CART: 'CLEAR_CART'
}

export const actionTypes = {
  GET_CHECKOUT_DATA: 'GET_CHECKOUT_DATA',
  GET_CART: 'GET_CART',
  GET_COUNTRIES: 'GET_COUNTRIES',
  GET_REGIONS: 'GET_REGIONS',
  GET_STATES: 'GET_STATES',
  GET_OFFICES: 'GET_OFFICES',
  ADD_ITEM: 'ADD_ITEM',
  UPDATE_ITEM: 'UPDATE_ITEM',
  REMOVE_ITEM: 'REMOVE_ITEM'
}

export const state = () => ({
  cartData: {},
  countries: [],
  regions: [],
  states: [],
  offices: [],
  isOrderSaved: false,
  isLoading: false
})

export const mutations = {
  [mutationTypes.SET_CART_DATA](state, cartData) {
    state.cartData = cartData
  },
  [mutationTypes.SET_LOADING_START](state) {
    state.isLoading = true
  },
  [mutationTypes.SET_LOADING_SUCCESS](state, { key, value }) {
    if (state.isLoading) {
      state.isLoading = false
    }

    state[key] = value
  },
  [mutationTypes.SET_COUNTRIES](state, countries) {
    state.countries = countries
  },
  [mutationTypes.SET_REGIONS](state, regions) {
    state.regions = regions
  },
  [mutationTypes.SET_STATES](state, states) {
    state.states = states
  },
  [mutationTypes.SET_OFFICES](state, offices) {
    state.offices = offices
  },
  [mutationTypes.SET_ORDER_SAVE](state) {
    state.isOrderSaved = true
    state.cartData = {}
  },
  [mutationTypes.CLEAR_CART](state) {
    state.cartData = {}
    state.isOrderSaved = false
  }
}

export const actions = {
  async [actionTypes.GET_CHECKOUT_DATA]({ commit }) {
    const STAGE_API_URL = 'https://back.virna.com.ua'
    const { data: checkoutData } = await this.$axios.$get(
      `${STAGE_API_URL}/api/checkout`
    )

    commit(mutationTypes.SET_COUNTRIES, checkoutData.countries)
    commit(mutationTypes.SET_REGIONS, checkoutData.regions)
  },
  async [actionTypes.GET_CART]({ commit }, cartId) {
    // const { data: cartData } = await this.$axios.$get('/api/checkout/cart')

    // FIXME: temporary solution
    const STAGE_API_URL = 'https://back.virna.com.ua'
    // const STAGE_API_URL = 'http://localhost:7000'
    // const { data: cartData } = await this.$axios.$get(
    //   `${STAGE_API_URL}/api/checkout/cart`
    // )

    const { data: cartData } = await this.$axios.$get(
      `${STAGE_API_URL}/api/checkout/cart/${cartId}`
    )

    if (cartData) {
      commit(mutationTypes.SET_CART_DATA, cartData)
    }
  },
  async [actionTypes.GET_REGIONS]({ commit }, regionId) {
    commit(mutationTypes.SET_LOADING_START)

    // FIXME: temporary solution
    const STAGE_API_URL = 'https://back.virna.com.ua'
    const { data: regions } = await this.$axios.$get(
      `${STAGE_API_URL}/api/shipping/regions`
    )

    commit(mutationTypes.SET_LOADING_SUCCESS, {
      key: 'regions',
      value: regions
    })
  },
  async [actionTypes.GET_STATES]({ commit }, regionId) {
    commit(mutationTypes.SET_LOADING_START)

    // FIXME: temporary solution
    const STAGE_API_URL = 'https://back.virna.com.ua'
    const { data: states } = await this.$axios.$get(
      `${STAGE_API_URL}/api/shipping/states?region_id=${regionId}`
    )

    commit(mutationTypes.SET_LOADING_SUCCESS, {
      key: 'states',
      value: states
    })
  },
  async [actionTypes.GET_OFFICES]({ commit }, stateId) {
    commit(mutationTypes.SET_LOADING_START)

    // FIXME: temporary solution
    const STAGE_API_URL = 'https://back.virna.com.ua'
    const { data: offices } = await this.$axios.$get(
      `${STAGE_API_URL}/api/shipping/offices?state_id=${stateId}`
    )

    commit(mutationTypes.SET_LOADING_SUCCESS, {
      key: 'offices',
      value: offices
    })
  },
  async [actionTypes.ADD_ITEM]({ commit }, { cartId, itemId }) {
    commit(mutationTypes.SET_LOADING_START)

    // FIXME: temporary solution
    // const { data: cartData } = await this.$axios.$post(
    //   `/api/checkout/cart/add/${id}`
    // )

    const STAGE_API_URL = 'https://back.virna.com.ua'
    // const STAGE_API_URL = 'http://localhost:7000'
    const { data: cartData } = await this.$axios.$post(
      `${STAGE_API_URL}/api/checkout/cart/add`,
      {
        cart: cartId,
        variant: itemId
      }
    )

    if (!state.cartData) {
      commit(
        `localStorage/${localStorageMutationTypes.SET_CART_ID}`,
        cartData.id,
        { root: true }
      )
    }

    commit(mutationTypes.SET_LOADING_SUCCESS, {
      key: 'cartData',
      value: cartData
    })
  },
  async [actionTypes.UPDATE_ITEM]({ commit }, { cartId, itemId, body }) {
    commit(mutationTypes.SET_LOADING_START)

    // FIXME: temporary solution
    const STAGE_API_URL = 'https://back.virna.com.ua'
    const { data: cartData } = await this.$axios.$put(
      `${STAGE_API_URL}/api/checkout/cart/${cartId}/update/${itemId}`,
      body
    )

    // const { data: cartData } = await this.$axios.$put(
    //   `/api/checkout/cart/update/${id}`,
    //   body
    // )

    commit(mutationTypes.SET_LOADING_SUCCESS, {
      key: 'cartData',
      value: cartData
    })
  },
  async [actionTypes.REMOVE_ITEM]({ commit }, { cartId, itemId }) {
    commit(mutationTypes.SET_LOADING_START)

    // FIXME: temporary solution
    const STAGE_API_URL = 'https://back.virna.com.ua'
    const { data: cartData } = await this.$axios.$delete(
      `${STAGE_API_URL}/api/checkout/cart/${cartId}/remove-item/${itemId}`
    )

    if (!cartData) {
      commit(`localStorage/${localStorageMutationTypes.SET_CART_ID}`, null, {
        root: true
      })
    }

    commit(mutationTypes.SET_LOADING_SUCCESS, {
      key: 'cartData',
      value: cartData || {}
    })
  }
}
