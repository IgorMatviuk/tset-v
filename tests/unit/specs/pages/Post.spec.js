import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'

import Post from '@/pages/-post'

import { mutations as rootMutations } from '@/store'

import createFactory from '@/api/helpers/factory'
import postAttrs from '@/api/factories/post'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('Post', () => {
  let app
  let route
  let store
  let post
  let relatedPosts
  let apiResponse
  let page

  function createPostPage(page = {}) {
    const wrapper = shallowMount(Post, {
      localVue,
      store,
      data() {
        return { page }
      },
      mocks: {
        $t: jest.fn(),
        $i18n: {
          path: (str) => str
        }
      }
    })

    return wrapper
  }

  beforeEach(() => {
    post = createFactory(postAttrs, 1)[0]
    relatedPosts = createFactory(postAttrs, 5)

    app = {
      i18n: {
        t: (str) => str,
        path: (str) => str
      },
      $axios: {
        $get: jest.fn().mockImplementation((url) => {
          if (url.includes('not-exist')) {
            return Promise.reject(new Error('Page not found'))
          } else {
            return Promise.resolve(apiResponse)
          }
        })
      }
    }

    route = {
      params: {
        slug: 'some-slug'
      }
    }

    store = new Vuex.Store({
      state: {
        locale: 'ru',
        fallbackLocale: 'ru',
        breadcrumb: []
      },
      mutations: {
        ...rootMutations
      }
    })

    apiResponse = {
      data: {
        post,
        related_posts: relatedPosts
      },
      links: {
        prev: {
          slug: '',
          title: ''
        },
        next: {
          slug: '',
          title: ''
        }
      },
      meta: {
        title: 'Пост Virna',
        description: 'Virna',
        keywords: 'Virna',
        og_title: 'Virna',
        og_image: '/img/some-img.jpg',
        og_description: 'Virna'
      }
    }

    page = {
      data: {
        post: {},
        related_posts: []
      },
      links: {
        prev: {
          slug: '',
          title: ''
        },
        next: {
          slug: '',
          title: ''
        }
      },
      meta: {
        title: '',
        description: '',
        keywords: '',
        og_title: '',
        og_image: '',
        og_description: ''
      }
    }
  })

  test('renders post page', () => {
    const wrapper = createPostPage(page)

    const { asyncData } = wrapper.vm.$options
    const ctx = { app, store, route }
    const data = { page: apiResponse }
    expect(asyncData(ctx)).resolves.toEqual(data)

    expect(wrapper.element).toMatchSnapshot()
  })

  test('calls error hook if page not found', async () => {
    route.params.slug = '/not-exist'

    const mockError = jest.fn()
    const mockRejectResult = { statusCode: 404, message: 'Page not found' }

    const wrapper = createPostPage(page)
    const { asyncData } = wrapper.vm.$options
    const ctx = { app, store, route, error: mockError }

    expect(asyncData(ctx)).resolves.toEqual({})
    await wrapper.vm.$nextTick()
    expect(mockError).toBeCalledWith(mockRejectResult)
  })
})
