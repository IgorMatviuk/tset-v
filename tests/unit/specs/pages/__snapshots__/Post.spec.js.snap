// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post renders post page 1`] = `
<main
  class="post-page"
>
  <thebreadcrumb-stub
    items="[object Object]"
  />
   
  <blockposthero-stub
    content="[object Object]"
  />
   
  <hr
    class="post-page__delimiter"
  />
    
  <blockpostrelatedslider-stub
    posts=""
  />
   
  <blockpostnavigation-stub
    links="[object Object]"
  />
</main>
`;
